src/CLI/Command/DebugExpression.php

```diff
@@ -0,0 +1,119 @@
+use Arkitect\Rules\ParsingError;
+use Arkitect\Rules\Violations;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DebugExpression extends Command
+{
+    /** @var string|null */
+    public static $defaultName = 'debug:expression';
+
+    /** @var string|null */
+    public static $defaultDescription = <<< 'EOT'
+Check which classes respect an expression
+EOT;
+
+    /** @var string */
+    public static $help = <<< 'EOT'
+Check which classes respect an expression
+EOT;
+
+    protected function configure(): void
+    {
+        $this
+            ->setHelp(self::$help)
+            ->addArgument('expression', InputArgument::REQUIRED)
+            ->addArgument('arguments', InputArgument::IS_ARRAY)
+            ->addOption(
+                'from-dir',
+                'd',
+                InputOption::VALUE_REQUIRED,
+                'The folder in which to search the classes',
+                '.'
+            )
+            ->addOption(
+                'target-php-version',
+                't',
+                InputOption::VALUE_OPTIONAL,
+                'Target php version to use for parsing'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $phpVersion = $input->getOption('target-php-version');
+        $targetPhpVersion = TargetPhpVersion::create($phpVersion);
+        $fileParser = FileParserFactory::createFileParser($targetPhpVersion);
+
+        $classSet = ClassSet::fromDir($input->getOption('from-dir'));
+        foreach ($classSet as $file) {
+            $fileParser->parse($file->getContents(), $file->getRelativePathname());
+            $parsedErrors = $fileParser->getParsingErrors();
+
+            if (\count($parsedErrors) > 0) {
+                $output->writeln('WARNING: Some files could not be parsed for these errors:');
+                /** @var ParsingError $parsedError */
+                foreach ($parsedErrors as $parsedError) {
+                    $output->writeln(' - '.$parsedError->getError().': '.$parsedError->getRelativeFilePath());
+                }
+                $output->writeln('');
+            }
+
+            $ruleName = $input->getArgument('expression');
+            /** @var class-string $ruleFQCN */
+            $ruleFQCN = 'Arkitect\Expression\ForClasses\\'.$ruleName;
+            $arguments = $input->getArgument('arguments');
+
+            try {
+                $expressionReflection = new \ReflectionClass($ruleFQCN);
+            } catch (\ReflectionException $exception) {
+                $output->writeln("Error: Expression '$ruleName' not found.");
+
+                return 2;
+            }
+
+            $constructorReflection = $expressionReflection->getConstructor();
```

Comment:
We can move these checks for arguments in a specific function to keep the main function cleaner, what do you think?

Reply:
Good catch, I extracted a method for this part and a couple of others, to simplify this huge method...